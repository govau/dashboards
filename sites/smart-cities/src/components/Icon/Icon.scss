@import '~style/main';

.svg {
  display: inline-block;
  vertical-align: middle;

  * {
    fill: currentColor;
  }
}

.indicatorMark {
  path { // the "C" or "P"
    fill: $color-white;
  }
}

.arrowInCircle {
  path {
    fill: none;
    stroke: $color-white;
  }
}

.arrowInCircleInverted {
  circle {
    fill: $color-white;
  }

  path {
    fill: none;
    stroke: currentColor;
  }
}

.sunSparkles {
  transform: rotate(0);
  transform-box: fill-box;
  transform-origin: center;
  transition: transform 500ms;

  svg:hover & {
    transform: rotate(90deg);
  }
}

// This will be applied to the SVG element itself
:global(.circleWithShadow) {
  @include shadow(1);

  border-radius: 50%;
  transition: box-shadow $animation-duration;

  &:hover {
    @include shadow(2);
  }
}
