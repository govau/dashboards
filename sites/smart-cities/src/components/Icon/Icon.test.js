import React from 'react';
import renderer from 'react-test-renderer';
import Icon from './Icon';

const defaultProps = {
  icon: 'questionMark',
};

it('should match Snapshot', () => {
  const renderedValue = renderer.create(<Icon {...defaultProps} />).toJSON();
  expect(renderedValue).toMatchSnapshot();
});
