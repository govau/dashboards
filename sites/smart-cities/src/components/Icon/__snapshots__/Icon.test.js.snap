// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
<svg
  className="circleWithShadow svg"
  height={null}
  style={
    Object {
      "color": undefined,
    }
  }
  viewBox="0 .5 25 25"
  width={null}
>
  <title>
    Question mark
  </title>
  <path
    d="M12.5,2 C6.43398605,2 1.5,6.93063282 1.5,12.9926003 C1.5,19.0545549 6.43398605,23.9851929 12.5,23.9851929 C18.5660651,23.9851929 23.5,19.0545549 23.5,12.9926003 C23.5,6.93063793 18.5660651,2 12.5,2 Z M12.5,3.53385013 C17.7365628,3.53385013 21.9651163,7.75958678 21.9651163,12.9926003 C21.9651163,18.2256112 17.7365628,22.4513427 12.5,22.4513427 C7.26348837,22.4513427 3.03488372,18.2255856 3.03488372,12.9926003 C3.03488372,7.7595919 7.26348837,3.53385013 12.5,3.53385013 Z M12.5,5.83462532 C10.3589395,5.83462532 8.60945349,7.22828155 8.17515814,8.966236 C8.09557133,9.23596721 8.17005237,9.52760137 8.36924882,9.72620755 C8.56844526,9.92481373 8.86041252,9.99854199 9.13008838,9.91833655 C9.39976424,9.83813111 9.60390151,9.6168543 9.66205116,9.34170973 C9.89530233,8.40848975 11.0561605,7.36847545 12.5,7.36847545 C13.2112651,7.36847545 13.9562209,7.62948562 14.4825581,8.05551249 C15.0089209,8.48153936 15.3139535,9.03014643 15.3139535,9.66925065 C15.3139535,10.508599 15.1428651,10.8853765 14.9142442,11.1791344 C14.6856488,11.4729178 14.3201674,11.7013081 13.8430233,12.0419251 C13.3658791,12.3825676 12.7991233,12.8528665 12.3880814,13.6157269 C11.9771163,14.3785872 11.7325581,15.3999013 11.7325581,16.8272256 C11.7325582,17.2508517 12.0761536,17.5942685 12.5,17.5942685 C12.9238464,17.5942685 13.2674418,17.2508517 13.2674419,16.8272256 C13.2674419,15.5703121 13.4786674,14.8261391 13.7391116,14.3426951 C13.9995302,13.8593278 14.3202953,13.5946364 14.7383721,13.2961748 C15.1564233,12.9977131 15.6865721,12.6908152 16.1293605,12.1218131 C16.5721744,11.5528058 16.8488372,10.747215 16.8488372,9.66925065 C16.8488372,8.51886305 16.2578047,7.52565674 15.4418605,6.86518088 C14.6258651,6.20473058 13.5771047,5.83462532 12.5,5.83462532 L12.5,5.83462532 Z M12.5079302,18.105434 C11.9423512,18.105434 11.476693,18.5632627 11.476693,19.1280008 C11.476693,19.6927388 11.9423767,20.1505675 12.5079302,20.1505675 C13.0734837,20.1505675 13.531186,19.6927388 13.531186,19.1280008 C13.531186,18.5632627 13.0734581,18.105434 12.5079302,18.105434 Z"
  />
</svg>
`;
